Source/Dafny/DafnyPrelude.bpl

```diff
@@ -323,6 +323,12 @@ function  $IsAlloc<T>(T,Ty,Heap): bool uses {
             $IsAllocBox(bx, t0, h)));
 }
 
+function $AlwaysAllocated(Ty): bool uses {
+    axiom (forall ty: Ty :: { $AlwaysAllocated(ty) }
+      $AlwaysAllocated(ty) ==>
+      (forall h: Heap, v: Box  :: { $IsAllocBox(v, ty, h) }  $IsBox(v, ty) ==> $IsAllocBox(v, ty, h)));
+}
```

Comment:
With the current design, each `$AlwaysAllocated` predicate will "immediately" give rise to L329, and L329 will be have as its trigger an `$IsAllocBox` for the always-allocated type. This seems good to me, because each method/function/other-large-proof-context will have a fixed number of always-allocated types, so "immediately" dismantling them into their own L329 seems appropriate.